L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -5,105 +5,174 @@
 
 HGCalConcentratorSuperTriggerCellImpl::
 HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf)
-  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize"))
+  : fixedDataSizePerHGCROC_(conf.getParameter<bool>("fixedDataSizePerHGCROC")),
+    coarseTCmapping_(conf)
 {
 
-    if ( stcSize_.size() != kNLayers_ ){
-        throw cms::Exception("HGCTriggerParameterError")
-            << "Inconsistent size of super trigger cell size vector" << stcSize_.size() ;
-    }
-    for(auto stc : stcSize_) {
-        if ( stc!=kSTCsizeFine_ && stc!=kSTCsizeCoarse_ ){
-            throw cms::Exception("HGCTriggerParameterError")
-              << "Super Trigger Cell should be of size "<<
-              kSTCsizeFine_ << " or " << kSTCsizeCoarse_;
-        }
-    }
-    
-}
+    // if ( stcSize_.size() != kNLayers_ ){
+    //     throw cms::Exception("HGCTriggerParameterError")
+    //         << "Inconsistent size of super trigger cell size vector" << stcSize_.size() ;
+    // }
+    // for(auto stc : stcSize_) {
 
-std::map<int,int> 
-HGCalConcentratorSuperTriggerCellImpl::kSplit_ = {
-  {kSTCsizeFine_, kSplit_v8_Fine_},
-  {kSTCsizeCoarse_, kSplit_v8_Coarse_}
-};
+    //   coarseTCmapping_.checkSizeValidity(stc);
 
-int
-HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
+    // }
+  //    coarseTCmapping_(conf);
+    std::string energyType(conf.getParameter<string>("type_energy_division"));
 
+    if( energyType == "superTriggerCell" ){
+      energyDivisionType_ = superTriggerCell;
+    }else if( energyType == "oneBitFraction" ){
+      energyDivisionType_ = oneBitFraction;
 
-  DetId TC_id( detid );
-  if ( TC_id.det() == DetId::Forward ){//V8
+      oneBitFractionThreshold_ = conf.getParameter<double>("oneBitFractionThreshold");
+      oneBitFractionLowValue_ = conf.getParameter<double>("oneBitFractionLowValue");
+      oneBitFractionHighValue_ = conf.getParameter<double>("oneBitFractionHighValue");
 
-    HGCalDetId TC_idV8(detid);
+    }else if( energyType == "equalShare" ){
+      energyDivisionType_ = equalShare;
 
-    if( triggerTools_.isScintillator(detid) ){
-      return TC_idV8.cell(); //scintillator
-    }
-    else{
-      int TC_wafer = TC_idV8.wafer();
-      int thickness = triggerTools_.thicknessIndex(detid,true);
-      int TC_split = ( TC_idV8.cell() & kSplit_.at( stcSize_.at(thickness) ) );
+      nTriggerCellsForDivision_ = conf.getParameter<int>("nTriggerCellsForDivision");
```

Comment:
the problem is that (currently) the STC size needs to be set to 8 for the fixed-data-per-hgcroc scheme in the low-density modules, in order to pick up the correct TCs. However there are actually only 4 "coarse-2" TCs.
I could set this to be calculated to be equal to the size of the STC if not fixed-data-per-hgcroc and 4 otherwise perhaps.

Reply:
Yes, it's probably better to deal with that internally to avoid parameter inconsistencies